### DISCUSSION TITLE
Unsigned Plugins

### DISCUSSION THREAD

**Discussion by alpacc** [Floorp-Projects/Floorp#1243] [Feature Request]
TLDR: Firefox uninstalls unsigned XPI files after browser restart and I want it to not do that.

Longer:
In chrome or in some other forks of firefox(including the developer version), it is possible to add any extension/plugin straight from the source code. In Floorp, and therefore default Firefox, it lets you install them as "temporary plugin"s but that's it. 
Keeping this security feature how it is by default makes sense but I develop extensions for myself to use and it feels controlling/restrictive not being able to keep them installed without giving firefox my code and having them tell me its ok to install my own extension. I would like extensions to be installable without their permission. A button or even just a flag somewhere would save me time and make me feel more "free" using the browser. Thank you for the incredible work!

**Comment by stilgarwolf** [comment]
I'm using my own unsigned extensions: Extensions->Manage extensions->Tool for all add-ons->Install Add-on From File
let you install your unsigned zipped extension.
The extension is not discarded at next run.

**Reply by alpacc** [reply]
Thank you for the reply!
That is how I was doing it before. For some reason though now that I went in to get screenshots it is doing something different. Now it says it is corrupt, but I know it is not corrupt because it is the same file I uploaded and got signed and have running right now. I also tried a nearly blank example one and it called that corrupt as well.

Are you sure the one you are trying isn't signed? 

![O9PZ6YT](https://github.com/Floorp-Projects/Floorp/assets/117709578/47fa0a8e-8759-4be4-9a90-3f42634da0ee)

**Reply by stilgarwolf** [reply]
Unsigned. Just 2 files zipped: "my_code.js" and "manifest.json", no subdirs.
